app/routes/create-catch-certificate/$catchCertificate/what-are-you-exporting.tsx

```diff
@@ -14,7 +18,46 @@ export const loader: LoaderFunction = async ({ params, request }: DataFunctionAr
   return json(await getAddSpeciesLoaderData(catchCertificate, species, state, presenttion));
 };
 
+export const action: ActionFunction = async ({ request, params }): Promise<Response> => {
+  const { catchCertificate } = params;
+  const form = await request.formData();
+  const species: string = form.get("species") as string || '';
+  const state: string = form.get("state") as string || '';
+  const presentation: string = form.get("presentation") as string || '';
+  const commodityCode: string = form.get("commodityCode") as string || '';
+
+  const resquestBody: any = {
+    "addToFavourites": false,
+    "btn_submit":"",
+    "presentation": presentation,
+    "presentationLabel":"",
+    "redirect": `/create-catch-certificate/${catchCertificate}/what-are-you-exporting`,
+    "species":"",
+    "speciesCode": species,
+    "state": state,
+    "stateLabel":"",
+    "scientificName":"",
+    "commodity_code": commodityCode,
+    "commodity_code_description":""
+  }
+  
+  const response = await addSpecies(catchCertificate, resquestBody);
+  const errors: IError[] = response.errors || [];
```

Comment:
I think I start seeing this in more than one place... shall we put in a common place?

Reply:
Updated @daniele-zurico, we now have this function `export const apiCallFailed: (errors: IError[], data?: any) => Response` to centralise this behaviour.